pkgs/by-name/nc/nchat/package.nix

```diff
@@ -0,0 +1,75 @@
+{
+  lib,
+  fetchFromGitHub,
+  file, # for libmagic
+  ncurses,
+  openssl,
+  readline,
+  sqlite,
+  zlib,
+  cmake,
+  gperf,
+  stdenv,
+  darwin,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "nchat";
+  version = "5.2.11";
+
+  src = fetchFromGitHub {
+    owner = "d99kris";
+    repo = "nchat";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-LcTVftLKlzHstSaJjdtqoEmQ7kiqft9dbXRXoYjbaus=";
+  };
+
+  postPatch = ''
+    substituteInPlace lib/tgchat/ext/td/CMakeLists.txt \
+      --replace "get_git_head_revision" "#get_git_head_revision"
+    substituteInPlace lib/tgchat/CMakeLists.txt \
+      --replace-fail "list(APPEND OPENSSL_ROOT_DIR" "#list(APPEND OPENSSL_ROOT_DIR"
+    substituteInPlace src/main.cpp \
+      --replace-fail 'FileUtil::DirName(FileUtil::GetSelfPath()) + "/../" ' ""
```

Comment:
Does putting `"-DCMAKE_INSTALL_LIBDIR=lib"` in `cmakeFlags` work instead of this patch?

Reply:
Looking at the code, it should work just fine. In fact, that way it's cleaner that trying to modify the source code.
I'll give it a try to confirm and if it works as expected, I'll push the modifications.